src/sql/workbench/contrib/notebook/browser/notebookLinkHandler.ts

```diff
@@ -118,6 +118,20 @@ export class NotebookLinkHandler {
 		}
 	}
 
+	public getEncodedLinkUrl(linkUrl: string): string {
+		// Need to encode URI here in order for user to click the proper encoded link in WYSIWYG
+		// skip encoding it if it's already encoded
+		if (!this.containsEncodedComponents(linkUrl)) {
+			return encodeURI(linkUrl);
+		}
+		return linkUrl;
+	}
+
+	public containsEncodedComponents(url) {
```

Comment:
Please add a doc comment explaining what this is doing (for example - what is a "component" in this case)